File: app/src/main/java/com/elvotra/clean/presentation/ui/adapters/PostsRecyclerAdapter.java
package com.elvotra.clean.presentation.ui.adapters;

import android.support.v7.widget.RecyclerView;
import android.text.Html;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.elvotra.clean.R;
import com.elvotra.clean.presentation.model.PostViewItem;
import com.squareup.picasso.Picasso;

import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;

public class PostsRecyclerAdapter
        extends RecyclerView.Adapter<PostsRecyclerAdapter.ViewHolder> {

    public interface MovieListItemClickListener {

        void onMovieClicked(long movieId);

    }

    private List<PostViewItem> postViewItems;
    private PostsRecyclerAdapter.MovieListItemClickListener listener;

    public class ViewHolder extends RecyclerView.ViewHolder
            implements View.OnClickListener {

        @BindView(R.id.list_item_post_user_avatar)
        ImageView userAvatar;
        @BindView(R.id.list_item_post_title)
        TextView postTitle;
        @BindView(R.id.list_item_post_body)
        TextView postBody;
        @BindView(R.id.list_item_post_user)
        TextView postUser;
        @BindView(R.id.list_item_post_comments_count)
        TextView postCommentsCount;

        public ViewHolder(View item) {
            super(item);

            ButterKnife.bind(this, item);
            item.setOnClickListener(this);

        }

        @Override
        public void onClick(View view) {

            listener.onMovieClicked(postViewItems.get(getAdapterPosition()).getId());

        }

    }

    public PostsRecyclerAdapter(List<PostViewItem> postViewItems,
                                PostsRecyclerAdapter.MovieListItemClickListener listener) {

        this.postViewItems = postViewItems;
        this.listener = listener;

    }

    @Override
    public PostsRecyclerAdapter.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View v = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.list_item_post, parent, false);

        PostsRecyclerAdapter.ViewHolder vh = new PostsRecyclerAdapter.ViewHolder(v);

        return vh;

    }

    @Override
    public void onBindViewHolder(PostsRecyclerAdapter.ViewHolder holder, int position) {

        Picasso.get()
                .load(postViewItems.get(position).getAvatar())
                .resize(100, 150)
                .centerCrop()
                .into(holder.userAvatar);
        holder.postTitle.setText(Html.fromHtml(postViewItems.get(position).getTitle()));
        holder.postBody.setText(postViewItems.get(position).getBody());
        holder.postUser.setText(postViewItems.get(position).getUser());
        holder.postCommentsCount.setText("" + postViewItems.get(position).getCommentsCount());

    }

    @Override
    public int getItemCount() {

        return postViewItems.size();

    }

}

File: app/src/main/java/com/elvotra/clean/presentation/ui/adapters/PostsRecyclerAdapter.kt
package com.elvotra.clean.presentation.ui.adapters

import android.content.Context
import android.support.v7.widget.RecyclerView
import android.text.Html
import android.text.TextUtils
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import com.bumptech.glide.Glide
import com.bumptech.glide.request.RequestOptions.circleCropTransform
import com.elvotra.clean.R
import com.elvotra.clean.presentation.model.PostViewItem

class PostsRecyclerAdapter(private val context: Context, private val postViewItems: List<PostViewItem>,
                           private val listener: PostsListItemClickListener) : RecyclerView.Adapter<PostsRecyclerAdapter.ViewHolder>() {

    interface PostsListItemClickListener {

        fun onPostClicked(postId: Int)

    }

    inner class ViewHolder(item: View) : RecyclerView.ViewHolder(item), View.OnClickListener {

        var list_item_post_user_avatar: ImageView? = null
        var list_item_post_title: TextView? = null
        var list_item_post_body: TextView? = null
        var list_item_post_user: TextView? = null
        var list_item_post_comments_count: TextView? = null

        init {

            item.setOnClickListener(this)

        }

        override fun onClick(view: View) {

            listener.onPostClicked(postViewItems[adapterPosition].id)

        }

    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): PostsRecyclerAdapter.ViewHolder {
        val v = LayoutInflater.from(parent.context)
                .inflate(R.layout.list_item_post, parent, false)

        return ViewHolder(v)

    }

    override fun onBindViewHolder(holder: PostsRecyclerAdapter.ViewHolder, position: Int) {
        Glide.with(context)
                .load(postViewItems[position].avatar)
                .apply(circleCropTransform())
                .into(holder.list_item_post_user_avatar!!)
        holder.list_item_post_title!!.text = Html.fromHtml(postViewItems[position].title)
        holder.list_item_post_body!!.text = postViewItems[position].body
        holder.list_item_post_user!!.text = postViewItems[position].user
        val commentsCount = postViewItems[position].commentsCount
        val comments = if (TextUtils.isEmpty(commentsCount)) context.getString(R.string.no_comments) else context.getString(R.string.posts_comments, commentsCount)
        holder.list_item_post_comments_count!!.text = comments
    }

    override fun getItemCount(): Int {
        return postViewItems.size
    }
}



